cmake/modules/SIMDExt.cmake

```diff
@@ -18,23 +18,9 @@ if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|AARCH64")
   set(HAVE_ARM 1) 
   set(save_quiet ${CMAKE_REQUIRED_QUIET})
   set(CMAKE_REQUIRED_QUIET true)
-  include(CheckCXXSourceCompiles)
-
-  check_cxx_source_compiles("
-    #define CRC32CX(crc, value) __asm__(\"crc32cx %w[c], %w[c], %x[v]\":[c]\"+r\"(crc):[v]\"r\"(value))
-    asm(\".arch_extension crc\");
-    unsigned int foo(unsigned int ret) {
-      CRC32CX(ret, 0);
-      return ret;
-    }
-    int main() { foo(0); }" HAVE_ARMV8_CRC)
-    check_cxx_source_compiles("
-    asm(\".arch_extension crypto\");
-    unsigned int foo(unsigned int ret) {
-      __asm__(\"pmull  v2.1q,          v2.1d,  v1.1d\");
-      return ret;
-    }
-    int main() { foo(0); }" HAVE_ARMV8_CRYPTO)
+  include(CheckCCompilerFlag)
+  CHECK_C_COMPILER_FLAG(-march=armv8-a+crc HAVE_ARMV8_CRC)
+  CHECK_C_COMPILER_FLAG(-march=armv8-a+crypto HAVE_ARMV8_CRYPTO)
```

Comment:
Does that mean we implicitly add the -march? We don't want to set -march=+crc/crypto because that could potentially generate invalid code for plain v8.0 compatible CPUs.

Reply:
no, this does not the imply `-march` option. we pass the `-march` option to the compiler via `${ARMV8_CRC_COMPILE_FLAGS}`, which only applies to crc32 code.